src/utils/gitUtils.ts

```diff
@@ -0,0 +1,145 @@
+        gitWorkspaceState.hasAdminAccess = originUrl ? hasAdminAccessToRepo(await (tryGetReposGetResponseData(context, originUrl))) : false;
+    }
+
+    return gitWorkspaceState;
+}
+
+/* Function used to enforce all our opinionated requirements for creating a Static Web App */
+export async function verifyGitWorkspaceForCreation(context: IActionContext, gitWorkspaceState: GitWorkspaceState, uri: Uri): Promise<VerifiedGitWorkspaceState> {
+    const gitFolderName: string = '.git';
+    if ((await fse.readdir(uri.fsPath)).filter(file => file !== gitFolderName).length === 0) {
+        // if the git repo is empty, git push will fail and create an empty GitHub repo so throw an error here
+        throw new Error(localize('emptyWorkspace', 'Cannot create a Static Web App with an empty workspace.'));
+    }
+
+    if (!gitWorkspaceState.repo) {
+        const gitRequired: string = localize('gitRequired', 'A GitHub repository is required to create a Static Web App. Would you like to initialize your project as a git repository and create a GitHub remote?');
+        context.telemetry.properties.cancelStep = 'initRepo';
+
+        await ext.ui.showWarningMessage(gitRequired, { modal: true }, DialogResponses.yes);
+        const gitApi: API = await getGitApi();
+        let newRepo: Repository | null = await gitApi.init(uri);
+        if (!newRepo) {
+            const localGit: git.SimpleGit = git(uri.fsPath);
+            await localGit.init({});
+            newRepo = await gitApi.openRepository(uri);
+            if (!newRepo) {
+                throw new Error(localize('gitInitFailed', 'Git initialization failed.  Please initialize a git repository manually and attempt to create again.'));
+            }
+
+        }
+
+        await promptForCommit(newRepo, localize('initCommit', 'Initial commit'));
+        gitWorkspaceState.repo = newRepo;
+
+    } else if (!!gitWorkspaceState.remoteUrl && !gitWorkspaceState.hasAdminAccess) {
+        context.telemetry.properties.cancelStep = 'adminAccess';
+
+        const adminAccess: string = localize('adminAccess', 'Admin access to the GitHub repository is required.  Please use a repo with admin access or create a fork.');
+        await ext.ui.showWarningMessage(adminAccess, { modal: true });
+
+    } else if (gitWorkspaceState.dirty && gitWorkspaceState.repo) {
+        context.telemetry.properties.cancelStep = 'dirtyWorkspace';
+
+        const commitChanges: string = localize('commitChanges', 'Commit all working changes to create a Static Web App.');
+        await ext.ui.showWarningMessage(commitChanges, { modal: true }, { title: localize('commit', 'Commit') });
+        await promptForCommit(gitWorkspaceState.repo, localize('commitMade', 'Commit made from VS Code Azure Static Web Apps'));
+        gitWorkspaceState.dirty = false;
+    }
+
+    return <VerifiedGitWorkspaceState>gitWorkspaceState;
+}
+
+export async function promptForCommit(repo: Repository, value?: string): Promise<void> {
```

Comment:
nit: No need to export. Actually does it really matter too much if we export something that doesn't need to be exported?

Reply:
Probably not.  It does make it harder to determine if we have unused code.  If something is exported, no error is thrown if a function isn't being used.  

I guess also if I didn't want it accessible outside of the file, that'd obviously be a problem.